src/cli/config-flags.ts

```diff
@@ -0,0 +1,137 @@
+import type { LogLevel, TaskCommand } from '@stencil/core/declarations';
```

Comment:
This PR creates a new module here which exports readonly arrays of all our supported CLI args and uses those to programmatically construct an interface (`ConfigFlags`) into which all of our CLI args get parsed. Because this is created on the basis of readonly arrays of strings, we have access to those keys both at runtime _and_ in the type system, so we can use them to write our actual arg parsing code and to create a type which describes the object we're going to parse them into. This lets us write a given flag down only in one place and have some consistency in the world.